Extension/src/Debugger/configurationProvider.ts

```diff
@@ -7,11 +7,73 @@ import * as debugUtils from './utils';
 import * as os from 'os';
 import * as path from 'path';
 import * as vscode from 'vscode';
+import { getBuildTasks, BuildTaskDefinition } from '../LanguageServer/extension';
+import * as util from '../common';
+import * as fs from 'fs';
+import * as assert from 'assert';
+import * as Telemetry from '../telemetry';
 
 import { IConfiguration, IConfigurationSnippet, DebuggerType, MIConfigurations, WindowsConfigurations, WSLConfigurations, PipeTransportConfigurations } from './configurations';
 import { parse } from 'jsonc-parser';
+import { PlatformInformation } from '../platform';
+
+/*
+ * Retrieves configurations from a provider and displays them in a quickpick menu to be selected.
+ * Ensures that the selected configuration's preLaunchTask (if existent) is populated in the user's task.json.
+ * Automatically starts debugging for "Build and Debug" configurations.
+ */
+export class QuickPickConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private underlyingProvider: vscode.DebugConfigurationProvider;
+
+    public constructor(provider: CppConfigurationProvider) {
+        this.underlyingProvider = provider;
+    }
+
+    async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
+        const configs: vscode.DebugConfiguration[] = await this.underlyingProvider.provideDebugConfigurations(folder, token);
+        
+        const editor: vscode.TextEditor = vscode.window.activeTextEditor;
+        if (!editor || !util.fileIsCOrCppSource(editor.document.fileName) || configs.length <= 1) {
+            const defaultConfig: vscode.DebugConfiguration = configs.find(config => { return config.name === "(gdb) Launch" || config.name === "(lldb) Launch"; });
+            assert(defaultConfig);
```

Comment:
I don't think we want to use `assert` in the extension.  If I recall correctly, `assert` will terminate the extension process if the condition fails.